crates/next-core/src/next_config.rs

```diff
@@ -1395,11 +1416,12 @@ impl NextConfig {
             return Ok(Vc::cell(Vec::new()));
         }
         let mut rules = Vec::new();
-        for (glob, rule) in turbo_rules.iter() {
-            fn transform_loaders(loaders: &[LoaderItem]) -> ResolvedVc<WebpackLoaderItems> {
+        for (glob, rule_collection) in turbo_rules.iter() {
+            fn transform_loaders(
+                loaders: Box<dyn Iterator<Item = &LoaderItem> + '_>,
```

Comment:
The Box seems pretty weird here...

```suggestion
                loaders: &mut dyn Iterator<Item = &LoaderItem>,
```

Reply:
Good call.